### ISSUE TITLE
Very shallow search when MultiPV greater than the number of legal moves.

### ISSUE THREAD

**Issue by dsjoerg** [jdart1/arasan-chess#14]
I was surprised that when running with MultiPV greater than the number of legal moves, the search does not appear to run to the desired depth.

Example:

```
Arasan v20.4.1-13-g0c84328 Copyright 1994-2018 by Jon Dart. All Rights Reserved.
uci
id name Arasan v20.4.1-13-g0c84328
...
setoption name MultiPV value 10
position fen r1bqkb1r/ppp2ppp/5n2/nB1Pp1N1/8/8/PPPP1PPP/RNBQK2R b KQkq - 2 6
go depth 12
failed to open ../bin/book.bin
info multipv 1 depth 1 score cp -50 time 0 nodes 26 hashfull 0 pv c7c6
info multipv 2 depth 1 score cp -74 time 0 nodes 32 hashfull 0 pv c8d7
info multipv 3 depth 1 score cp -173 time 0 nodes 37 hashfull 0 pv e8e7
info multipv 4 depth 1 score cp -221 time 0 nodes 41 hashfull 0 pv f6d7
info multipv 5 depth 1 score cp -363 time 0 nodes 44 hashfull 0 pv a5c6 d5c6
info multipv 6 depth 1 score cp -689 time 0 nodes 44 hashfull 0 pv d8d7 b5d7
info multipv 1 depth 1 score cp -50 time 0 nodes 26 hashfull 0 pv c7c6
info multipv 2 depth 1 score cp -74 time 0 nodes 32 hashfull 0 pv c8d7
info multipv 3 depth 1 score cp -173 time 0 nodes 37 hashfull 0 pv e8e7
info multipv 4 depth 1 score cp -221 time 0 nodes 41 hashfull 0 pv f6d7
info multipv 5 depth 1 score cp -363 time 0 nodes 44 hashfull 0 pv a5c6 d5c6
info multipv 6 depth 1 score cp -689 time 0 nodes 46 hashfull 0 pv d8d7 b5d7
bestmove c7c6
```

Whereas if we run with a lower MultiPV setting, we get much more reasonable output:

```
Arasan v20.4.1-13-g0c84328 Copyright 1994-2018 by Jon Dart. All Rights Reserved.
uci
id name Arasan v20.4.1-13-g0c84328
...
setoption name MultiPV value 5
position fen r1bqkb1r/ppp2ppp/5n2/nB1Pp1N1/8/8/PPPP1PPP/RNBQK2R b KQkq - 2 6
go depth 12
<output is normal, ending with:>
info multipv 1 depth 12 score cp -46 time 241 nodes 292523 nps 1213788 hashfull 0 pv c7c6 d5c6
info multipv 2 depth 12 score cp -61 time 241 nodes 292765 nps 1214792 hashfull 0 pv c8d7 d1e2
info multipv 3 depth 12 score cp -153 time 244 nodes 295944 nps 1212885 hashfull 0 pv f6d7 g5e6 f7e6 d5e6 c7c6 e6d7 c8d7 d1h5 g7g6 h5e5 d8e7 e5e7 f8e7 b5e2 e8g8 d2d3
info multipv 4 depth 12 score cp -267 time 245 nodes 297422 nps 1213967 hashfull 0 pv e8e7 d1e2 d8d5 b1c3 d5g2 e2e5 c8e6 e5c7 f6d7 g5e6 g2h1 e1e2 f7e6 b5d7 e7f7 c7a5
info multipv 5 depth 12 score cp -549 time 257 nodes 312507 nps 1215980 hashfull 0 pv a5c6 d5c6 b7c6 b5c6 c8d7 d1f3 d7c6 f3c6 f6d7 c6d5 d8g5 d5a8 g5d8 a8c6 f8c5
bestmove c7c6 ponder d5c6
```

I can work around this behavior by setting MultiPV to be <= the number of legal moves, but I thought maybe there's something more fundamental I'm not getting, and perhaps this behavior is somehow correct?

Thanks in advance for any thoughts!

**Comment by jdart1** [comment]
Will look into it.

**Comment by jdart1** [comment]
Fixed in commit 72b9b88c26c4b7dca83188af91425711ab8c3e37

**Comment by dsjoerg** [comment]
Thank you!
